cmd/util/ledger/migrations/storage_v6.go

```diff
@@ -0,0 +1,3773 @@
+
+	m.Log.Info().Msg("Converting payloads...")
+
+	for _, payload := range storagePayloads {
+		keyParts := payload.Key.KeyParts
+		rawOwner := keyParts[0].Value
+		rawKey := keyParts[2].Value
+
+		err := m.reencodePayload(payload)
+
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to migrate key: %q (owner: %x): %w",
+				rawKey,
+				rawOwner,
+				err,
+			)
+		}
+
+		m.incrementProgress()
+	}
+	m.clearProgress()
+	m.Log.Info().Msg("Converting payloads complete")
+
+	// Encode the new values by calling `storage.Commit()`
+
+	m.Log.Info().Msg("Re-encoding converted values...")
+	m.incrementProgress()
+
+	err := m.storage.Commit(m.newInter, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to migrate payloads: %w", err)
+	}
+
+	m.completeProgress()
+
+	m.Log.Info().Msg("Re-encoding converted values complete")
+
+	m.clearProgress()
+	if m.emptyDeferredValues > 0 {
+		m.Log.Warn().Msgf("empty deferred values found: %d", m.emptyDeferredValues)
+	} else {
+		m.Log.Info().Msgf("empty deferred values found: %d", m.emptyDeferredValues)
+	}
+
+	m.clearProgress()
+	if m.skippedValues > 0 {
+		m.Log.Warn().Msgf("values not migrated: %d", m.skippedValues)
+	} else {
+		m.Log.Info().Msgf("values not migrated: %d", m.skippedValues)
+	}
+
+	return ledgerView.Payloads(), nil
+}
+
+func (m *StorageFormatV6Migration) incrementProgress() {
+	if m.progress == nil {
+		return
+	}
+
+	err := m.progress.Add(1)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (m *StorageFormatV6Migration) clearProgress() {
+	if m.progress == nil {
+		return
+	}
+
+	err := m.progress.Clear()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (m *StorageFormatV6Migration) completeProgress() {
+	if m.progress == nil {
+		return
+	}
+
+	if m.progress.IsFinished() {
+		return
+	}
+
+	err := m.progress.Finish()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (m *StorageFormatV6Migration) initPersistentSlabStorage(v *view) {
+	st := state.NewState(
+		v,
+		state.WithMaxInteractionSizeAllowed(math.MaxUint64),
+	)
+	stateHolder := state.NewStateHolder(st)
+	accounts := state.NewAccounts(stateHolder)
+
+	m.storage = runtime.NewStorage(
+		NewAccountsAtreeLedger(accounts),
+		func(f func(), _ func(metrics runtime.Metrics, duration time.Duration)) {
+			f()
+		},
+	)
+}
+
+func (m *StorageFormatV6Migration) getAccounts(payloads []ledger.Payload) state.Accounts {
+	l := NewView(payloads)
+	st := state.NewState(l)
+	sth := state.NewStateHolder(st)
+	accounts := state.NewAccounts(sth)
+	return accounts
+}
+
+func (m *StorageFormatV6Migration) getDeferredKeys(payloads []ledger.Payload) map[storagePath]bool {
+	m.clearProgress()
+	m.Log.Info().Msgf("Collecting deferred keys...")
+
+	deferredValuePaths := make(map[storagePath]bool)
+	for _, payload := range payloads {
+		m.incrementProgress()
+
+		keyParts := payload.Key.KeyParts
+		rawOwner := keyParts[0].Value
+		rawController := keyParts[1].Value
+		rawKey := keyParts[2].Value
+
+		if state.IsFVMStateKey(
```

Comment:
before calling this function the payloads were split: `fvmPayloads, storagePayloads, slabPayloads := splitPayloads(payloads)`

would it make sense here to only use ` storagePayloads` and `slabPayloads` here and skip this if condition?

Reply:
yeah, good idea!
We didn't have the split function earlier, and this wasn't cleaned up once it was introduced. good catch!